app/frontend/packs/nationalities-component.js

```diff
@@ -0,0 +1,118 @@
+    )
+  ) {
+    return;
+  }
+
+  addRemoveLink(
+    "candidate-interface-nationalities-form-other-nationality2-field"
+  );
+
+  addRemoveLink(
+    "candidate-interface-nationalities-form-other-nationality3-field"
+  );
+
+  addAddNationalityButton(
+    "#candidate-interface-nationalities-form-other-other-conditional"
+  );
+
+  hideSection(
+    "candidate-interface-nationalities-form-other-nationality2-field"
+  );
+
+  hideSection(
+    "candidate-interface-nationalities-form-other-nationality3-field"
+  );
+
+  function addRemoveLink(selector) {
+    const labelEl = document.querySelector(`[for=${selector}]`);
+    const removeLink = document.createElement("a");
+    removeLink.innerHTML = "Remove";
+    removeLink.classList.add("govuk-link", "remove-link");
+
+    // This has to be a link and not a button as the govuk-link class requires an
+    // href to apply  it's styling
+    removeLink.href = "#";
+    labelEl.appendChild(removeLink);
+
+    removeLink.addEventListener("click", function () {
+      addRemoveLinkEvent(labelEl, selector);
+    });
+  }
+
+  function addAddNationalityButton(parentSelector) {
+    const parent = document.querySelector(parentSelector);
+    const nationalityButton = document.createElement("button");
+    nationalityButton.innerHTML = "Add another nationality";
+    nationalityButton.id = "add-nationality-button";
+    nationalityButton.classList.add("govuk-button", "govuk-button--secondary");
+    parent.appendChild(nationalityButton);
+
+    const secondSelect = document.querySelector(
+      "[id=candidate-interface-nationalities-form-other-nationality2-field]"
+    );
+    const thirdSelect = document.querySelector(
+      "[id=candidate-interface-nationalities-form-other-nationality3-field]"
+    );
+
+    if (secondSelect.value && thirdSelect.value) {
+      nationalityButton.style.display = "none";
+    }
+
+    nationalityButton.addEventListener("click", function () {
+      event.preventDefault();
+      addNationalityEvent(nationalityButton);
+    });
+  }
+
+  function hideSection(selector) {
+    const select = document.getElementById(selector);
+    const labelEl = document.querySelector(`[for=${selector}]`);
+
+    if (select.value === "") {
+      labelEl.parentElement.style.display = "none";
+    }
+  }
+
+  function addRemoveLinkEvent(labelEl, selector) {
+    const addNationalityButton = document.getElementById(
+      "add-nationality-button"
+    );
+
+    addNationalityButton.style.display = "";
+    labelEl.parentElement.style.display = "none";
+    document.getElementById(selector).value = "";
+  }
+
+  function addNationalityEvent(nationalityButton) {
+
+    const secondFormLabel = document.querySelector(
+      "[for=candidate-interface-nationalities-form-other-nationality2-field]"
+    );
+    const thirdFormLabel = document.querySelector(
+      "[for=candidate-interface-nationalities-form-other-nationality3-field]"
+    );
```

Comment:
You could hoist all the `document.querySelector` calls to the start of this file, assign to variables, and then reuse those wherever you need later on:

```js
const secondFormLabel = document.querySelector(...)
const thirdFormLabel = document.querySelector(...)
let addNationalityButton = null

function addAddNationalityButton(...) {
  addNationalityButton = document.createElement("button")
  // ...
}

function addNationalityEvent(...) {
  if (secondFormLabel.parentElement.style.display === "none") {
    secondFormLabel.parentElement.style.display = "";
  } else {
    thirdFormLabel.parentElement.style.display = "";
  }
  // ...
}
```

This could shorten your functions and might make things a bit less repetitive. If that makes sense, do you want to give it a go?